nomos-da/network/messages/src/packing.rs

```diff
@@ -0,0 +1,156 @@
+// STD
+use futures::AsyncReadExt;
+use std::io;
+// Crates
+use nomos_core::wire;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+// Internal
+use crate::Result;
+
+type LenType = u16;
+const MAX_MSG_LEN_BYTES: usize = size_of::<LenType>();
+const MAX_MSG_LEN: usize = 1 << (MAX_MSG_LEN_BYTES * 8);
+
+fn into_failed_to_serialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to serialize message: {}", error),
+    )
+}
+
+fn into_failed_to_deserialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to deserialize message: {}", error),
+    )
+}
+
+fn get_message_too_large_error(message_len: usize) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!(
+            "Message too large. Maximum size is {}. Actual size is {}",
+            MAX_MSG_LEN, message_len
+        ),
+    )
+}
+
+pub fn pack<Message>(message: &Message) -> Result<Vec<u8>>
+where
+    Message: Serialize,
+{
+    let serialized_message = wire::serialize(message).map_err(into_failed_to_serialize)?;
+
+    let data_length = serialized_message.len();
+    if data_length > MAX_MSG_LEN {
+        return Err(get_message_too_large_error(data_length));
+    }
+
+    let mut buffer = Vec::with_capacity(MAX_MSG_LEN_BYTES + data_length);
+    buffer.extend_from_slice(&(data_length as u16).to_be_bytes());
```

Comment:
You have a `LenType` type alias declared on top, why use u16 here?

Reply:
Probably added the alias later, and forgot to update the references.